libslide/src/math/gcd_poly_zz.rs

```diff
@@ -0,0 +1,430 @@
+use crate::math::gcd;
+
+use core::cmp::{max, min};
+
+#[allow(unused)]
+type Poly = Vec<isize>;
+
+#[allow(unused)]
+type PolyPtrArg = [isize];
+
+#[allow(unused)]
+fn gcd_poly_zz_heu(f: Poly, g: Poly) -> Option<(Poly, Poly, Poly)> {
+    let res = trivial_gcd(&f, &g);
+    if res.is_some() {
+        return res;
+    }
+
+    let df = f.len();
+    let dg = f.len();
+
+    let (gcd_val, f, g) = poly_extract_common(&f, &g);
+
+    // hadnle single val case
+    if df == 1 || dg == 1 {
+        return Some((vec![], f, g));
+    }
```

Comment:
please make the common better here too. what does single val case mean? just that they have no degree?